lib/response.js

```diff
@@ -2,183 +2,187 @@
+    /* istanbul ignore if */
+    if (this._lightMyRequest.headers === null) {
+      this.copyHeaders()
+    }
+    this.serializeHeaders()
+    // Prepare response object
+    const res = {
+      raw: {
+        res: this,
+        req
+      },
+      headers: this._lightMyRequest.headers,
+      statusCode: this.statusCode,
+      statusMessage: this.statusMessage,
+      trailers: {},
+      get cookies () {
+        return setCookie.parse(this)
+      }
+    }
 
-  // Prepare payload parsers
-  res.json = function parseJsonPayload () {
-    return JSON.parse(res.payload)
-  }
+    // Prepare payload and trailers
+    const rawBuffer = Buffer.concat(this._lightMyRequest.payloadChunks)
+    res.rawPayload = rawBuffer
 
-  return res
-}
+    // we keep both of them for compatibility reasons
+    res.payload = rawBuffer.toString()
+    res.body = res.payload
+    res.trailers = this._lightMyRequest.trailers
 
-// Throws away all written data to prevent response from buffering payload
-function getNullSocket () {
-  return new Writable({
-    write (chunk, encoding, callback) {
-      setImmediate(callback)
+    // Prepare payload parsers
+    res.json = function parseJsonPayload () {
+      return JSON.parse(res.payload)
     }
-  })
-}
 
-function serializeHeaders (response) {
-  const headers = response._lightMyRequest.headers
+    return res
+  }
 
-  for (const headerName of Object.keys(headers)) {
-    const headerValue = headers[headerName]
-    if (Array.isArray(headerValue)) {
-      headers[headerName] = headerValue.map(value => '' + value)
-    } else {
-      headers[headerName] = '' + headerValue
+  /**
+   * @private
+   */
+  serializeHeaders () {
+    const headers = this._lightMyRequest.headers
+
+    for (const headerName of Object.keys(headers)) {
+      const headerValue = headers[headerName]
+      if (Array.isArray(headerValue)) {
+        headers[headerName] = headerValue.map(value => '' + value)
+      } else {
+        headers[headerName] = '' + headerValue
+      }
     }
   }
-}
-
-function copyHeaders (response) {
-  response._lightMyRequest.headers = Object.assign({}, response.getHeaders())
 
-  // Add raw headers
-  ;['Date', 'Connection', 'Transfer-Encoding'].forEach((name) => {
-    const regex = new RegExp('\\r\\n' + name + ': ([^\\r]*)\\r\\n')
-    const field = response._header.match(regex)
-    if (field) {
-      response._lightMyRequest.headers[name.toLowerCase()] = field[1]
-    }
-  })
+  /**
+   * @private
+   */
+  copyHeaders () {
+    this._lightMyRequest.headers = Object.assign({}, this.getHeaders())
+
+    // Add raw headers
+    ;['Date', 'Connection', 'Transfer-Encoding'].forEach((name) => {
+      // TODO change this to use the header getter
+      const regex = new RegExp('\\r\\n' + name + ': ([^\\r]*)\\r\\n')
+      const [, value] = this._header.match(regex) || []
+      // const value = this.getHeader(name)
+      if (value) {
+        this._lightMyRequest.headers[name.toLowerCase()] = value
```

Comment:
will the unit tests pass if you use the header getter?

Reply:
What you mean, this function same to original
What do you mean, this function is same as [original](https://github.com/fastify/light-my-request/blob/master/lib/response.js#L171-L182)

Reply:
Do not pass, these headers `'Data', 'Connection', 'Transfer-Encoding'` do not exist